### PR TITLE
Node 20.12.2 Iron (LTS)

### PR DESCRIPTION
**Pull request by arelra**
## What does this change?

Node 20.12.2

## Why?

Incorporate the latest bug fixes and updates:
https://github.com/nodejs/release#release-schedule

### CODE REVIEW

File: dotcom-rendering/scripts/check-node-versions.mjs [hunk 1]
```diff
@@ -39,7 +39,7 @@ const requiredNodeVersionMatches =
 		},
 		{
 			filepath: '../apps-rendering/riff-raff.yaml',
-			pattern: /^ +Recipe: .+-mobile-node(\d+\.\d+\.\d+).*$/m,
+			pattern: /^ +Recipe: apps-rendering.*-node-(\d+\.\d+\.\d+).*?$/m,
```
Comment (mxdvl) [inline dotcom-rendering/scripts/check-node-versions.mjs:42]: did you change the pattern in AMIgo?

**Review by mxdvl — APPROVED** [review]
That was easy, and we’re good [until April 2026](https://nodejs.org/en/about/previous-releases#release-schedule)!

🛠️ **Iron** ⚒️

File: dotcom-rendering/scripts/check-node-versions.mjs (hunk 1 above)
Comment (arelra) [reply dotcom-rendering/scripts/check-node-versions.mjs:42]: Yep: https://amigo.gutools.co.uk/recipes/apps-rendering-ARM-jammy-node-20.12.2/bakes/1

**Comment by arelra** [comment]
I think the Compressed Size workflow is failing because it is checking out a previous version which still requires 18.18.2 but actions/setup-node@v4 installs the latest version i.e. 20.12.2

File: scripts/env/check-node [hunk 2]
```diff
@@ -2,6 +2,15 @@
...
 
+# Check if we should skip this check
+# Required to prevent compressed-size-action@v2 from failing when on a branch that changes node versions
+# The action checks out main to compare against the current branch but does not reset the node version
+# check-node still runs as part of other CI jobs
+if [[ "$SKIP_CHECK_NODE" == "TRUE" ]]; then
+	echo "Skipping check-node"
+	exit 0
+fi
+
```
Comment (mxdvl) [inline scripts/env/check-node:13]: Is there a option to [update `compressed-size-action` upstream](https://github.com/preactjs/compressed-size-action/blob/master/action.yml)?

File: dotcom-rendering/package.json [hunk 3]
```diff
@@ -100,7 +100,7 @@
 		"@types/k6": "0.47.1",
 		"@types/lodash.debounce": "4.0.7",
 		"@types/lodash.get": "4.4.7",
-		"@types/node": "18.18.14",
+		"@types/node": "20.12.2",
```
Comment (mxdvl) [inline dotcom-rendering/package.json:103]: The [patch does not need to match the Node version](https://github.com/DefinitelyTyped/DefinitelyTyped#how-do-definitely-typed-package-versions-relate-to-versions-of-the-corresponding-library) and [should be 7](https://www.npmjs.com/package/@types/node/v/20.12.7).

**Review by mxdvl — APPROVED** [review]

File: apps-rendering/riff-raff.yaml [hunk 4]
```diff
@@ -12,7 +12,7 @@ templates:
             amiEncrypted: true
             amiTags:
                 # Keep the Node version in sync with `.nvmrc`
-                Recipe: jammy-mobile-node18.18.2-ARM
+                Recipe: apps-rendering-ARM-jammy-node-20.12.2
```
Comment (arelra) [inline apps-rendering/riff-raff.yaml:15]: The apps-rendering AMI name now aligns with the DCR naming convention

**Pull request merged**